src/App.tsx

```diff
@@ -14,11 +15,75 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string;
+};
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGood: 'Jam',
+  };
+
+  remover = (good: string) => {
+    if (this.state.selectedGood === good) {
+      return 'Romove';
+    }
+
+    return 'Select';
+  };
+
+  render(): React.ReactNode {
+    return (
+      <div className="content">
+        <h1 className="content__title">
+          {(this.state.selectedGood)
+            ? `Selected goods: ${this.state.selectedGood}`
+            : 'No goods selected'}
+        </h1>
+        {(this.state.selectedGood)
+          ? (
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ selectedGood: '' });
+              }}
+              className="button button__remove"
+            >
+              Clear
+            </button>
+          )
+          : ''}
+        <ul className="">
+          {goodsFromServer.map(good => (
+            <React.Fragment key={good}>
+              <div className="content__item">
+                <li
+                  className={(good === this.state.selectedGood)
+                    ? 'isSelected'
+                    : ''}
+                >
+                  {good}
+                </li>
+
+                <button
+                  type="button"
+                  onClick={() => {
+                    this.setState((state) => ({
+                      selectedGood: state.selectedGood === good ? '' : good,
+                    }));
+                  }}
+                  className={this.state.selectedGood === good
+                    ? 'button button__remove'
+                    : 'button'}
```

Comment:
Will be better to use classnames library
It's too easy to install it in your project and use
https://www.npmjs.com/package/classnames
1. `npm install classnames`
2. add this string to your App file `import classNames from 'classnames';`
3.  you can use `button` className as default and `button__remove` in condition
`className={classNames('button', 'button__remove' : this.state.selectedGood === good)}`